### ISSUE TITLE
Torch. float32 can only save 4 decimal places, but numpy can save 7 decimal places

### ISSUE THREAD

**Issue by Fightinging1997** [NVIDIA/DALI#4995]
### Version

1.28.0

### Describe the bug.

code:
from nvidia.dali.pipeline import pipeline_def
import nvidia.dali.types as types
import nvidia.dali.fn as fn
from nvidia.dali.plugin.pytorch import DALIGenericIterator

import time
start = time.time()
images_dir = "data/images_test"

@pipeline_def(num_threads=4, device_id=0)
def get_dali_pipeline():
    images, labels = fn.readers.file(
        file_root=images_dir, random_shuffle=True, name="Reader")
    # decode data on the GPU
    images = fn.decoders.image(
        images, device="mixed", output_type=types.RGB)

    images = fn.transpose(images, perm=[2, 0, 1])
    # TODO 无法变成float64, normalize会出错误, float64位后仍然保留出4位小数
    images /= 255.
    MEAN = types.Constant([0.30052053, 0.30516243, 0.30555184], shape=(3, 1, 1))
    STD = types.Constant([0.15082812, 0.14010814, 0.13871154], shape=(3, 1, 1))

    images = fn.normalize(images,
                 mean=MEAN,
                 stddev=STD)
    
    return images, labels

batch_size = 1
train_data = DALIGenericIterator(
    [get_dali_pipeline(batch_size=batch_size)],
    ['data', 'label'],
    reader_name='Reader'
)

for i, data in enumerate(train_data):
    x, y = data[0]['data'], data[0]['label']
print(x[0][0][0][-1])
print(x[0].dtype)
print((time.time() - start)/batch_size)

output:
tensor(-1.6805, device='cuda:0')
torch.float32

code2: 
image = cv2.imread("2.jpeg") # same image
image = np.ascontiguousarray(image.transpose(2, 0, 1))
image = image.astype(np.float32)
image /= 255.0
image -= np.array([0.30052053, 0.30516243,
                    0.30555184]).reshape(3, 1, 1)
image /= np.array([0.15082812, 0.14010814,
                    0.13871154]).reshape(3, 1, 1)
print(image[0][0][-1])
print(image.dtype)

output:
-1.6804672
dtype('float32')

This will result in a decrease in the accuracy of the model training process.

### Check for duplicates

- [X] I have searched the [open bugs/issues](https://github.com/NVIDIA/DALI/issues) and have found no duplicates for this bug report

**Comment by jantonguirao** [comment]
Hi @Fightinging1997. I believe this is a matter of printing only.
Please try to convert to numpy an then print:
```
# Your print
print(x[0][0][0][-1])
# Add this to check the precision is correct
print(x[0][0][0][-1].cpu().numpy())
```

**Comment by Fightinging1997** [comment]
Thanks
